pkgs/desktops/gnome-3/3.20/core/libpeas/default.nix

```diff
@@ -1,13 +1,15 @@
 { stdenv, fetchurl, pkgconfig, intltool, gnome3
-, glib, gtk3, gobjectIntrospection, python, pygobject3
+, glib, gtk3, gobjectIntrospection, python3, python3Packages, ncurses, python2, python2Packages
 }:
 
 stdenv.mkDerivation rec {
   inherit (import ./src.nix fetchurl) name src;
 
+  configureFlags = [ "--enable-python3" "--enable-python2" ];
+
   buildInputs =  [
-   intltool pkgconfig glib gtk3 gobjectIntrospection python pygobject3
-   gnome3.defaultIconTheme
+   intltool pkgconfig glib gtk3 gobjectIntrospection python3 python3Packages.pygobject3 python2
+   gnome3.defaultIconTheme ncurses
```

Comment:
I followed your advice but it doesn't work. Here is what the `libpeas/default.nix` looks like now:

```nix
let
  python2 = python2Packages.python.withPackages ( ps : with ps; [ pygobject3 ]);
  python3 = python3Packages.python.withPackages ( ps : with ps; [ pygobject3 ]);
in
stdenv.mkDerivation rec {
  inherit (import ./src.nix fetchurl) name src;

  configureFlags = [ "--enable-python3" "--enable-python2" ];

  buildInputs =  [
   intltool pkgconfig glib gtk3 gobjectIntrospection python2 python3
   gnome3.defaultIconTheme ncurses
  ];
  meta = [...];
}
```

When I build that, I get the following result:

```shellCommands
/nix/store/0xskayad86h730x6qaqzlyy8j8kalzlb-libpeas-1.18.0
+--- [...]
+---/nix/store/5wyc68zp1qj05ac5xgp18v8qpf259m96-python-2.7.11-env
|   +---/nix/store/dvq0mwb7ic8n6kxrqiykssqjq31gc3g4-pygobject-3.20.0
|   |   +--- [...]
+---/nix/store/nym478rhdc2ljk1pqfgic8cs7qh6a2wy-gobject-introspection-1.48.0 [...]
+---/nix/store/w9gwivwr7ih57b4y1dfhv64xy316p5qn-python3-3.5.1
[...]
```

As you can see, I get a `python-2.7.11-env` containing `pygobject` as a dependency, but the `python3-3.5.1` (without the `-env` suffix) does not depend on `pygobject`. When I launch `totem` which depends on this `libpeas`, I get an error message:

```shellCommands
$ totem
[...]
(totem:30126): libpeas-WARNING **: Error initializing Python Plugin Loader: PyGObject initialization failed
ImportError: could not import gobject (error was: ImportError("No module named 'gi'",))

(totem:30126): libpeas-WARNING **: Please check the installation of all the Python related packages required by libpeas and try again

(totem:30126): libpeas-WARNING **: Loader 'python3' is not a valid PeasPluginLoader instance
```

If I know check the deriver, I can confirm that my derivation seems fine and indeed includes `pygobject` for both python3 and python2:

```shellCommands
$ nix-store --query --deriver /nix/store/0xskayad86h730x6qaqzlyy8j8kalzlb-libpeas-1.18.0
/nix/store/3l1kj7wpby7gy5appwq99yc6bmfc4n5l-libpeas-1.18.0.drv
$ nix-store --query --references /nix/store/3l1kj7wpby7gy5appwq99yc6bmfc4n5l-libpeas-1.18.0.drv 
/nix/store/r13sd830c4576wr60n7inxl66qb7rn80-gobject-introspection-1.48.0.drv
/nix/store/m4dgc7jmg8mz85c4s13gh85icgac1r50-python3-3.5.1-env.drv
/nix/store/qq78rnjlcpj2g02gms4cjl2w68n6jd9i-python-2.7.11-env.drv
$ nix-store --query --references /nix/store/m4dgc7jmg8mz85c4s13gh85icgac1r50-python3-3.5.1-env.drv 
/nix/store/paq1akznhapmy1a4k66d95ip8zqv85rf-python3-3.5.1.drv
/nix/store/3cjzxahff1p406fkrfphh0b93sw7r89q-pygobject-3.20.0.drv
$ nix-store --query --references /nix/store/qq78rnjlcpj2g02gms4cjl2w68n6jd9i-python-2.7.11-env.drv 
/nix/store/a7lwpabsrajmkanhcxnkipf3ap6hjiwz-python-2.7.11.drv
/nix/store/zhajg8vxngjrzlp01rclh8ghrgvb5l8g-pygobject-3.20.0.drv
```